dex/src/main/scala/com/wavesplatform/dex/WavesDexCli.scala

```diff
@@ -176,6 +201,26 @@ object WavesDexCli {
                          |
                          |waves.dex.rest-api.api-key-hash = "$hashedApiKey"
                          |""".stripMargin)
+
+            case Command.CheckServer =>
+              (
+                for {
+                  _ <- tool.log(
+                    s"""
+                      |Passed arguments:
+                      |  Waves Node REST API  : ${args.nodeRestApi}
+                      |  Expected DEX version : ${args.version}
+                      |  DEX config path      : ${args.dexConfigPath}
+                   """.stripMargin
+                  )
+                  superConnector <- SuperConnector.create(args.dexConfigPath, args.nodeRestApi)
+                  checkResult    <- Checker(superConnector).checkState(args.version)
+                  _              <- tool.lift { superConnector.close() }
```

Comment:
Sadly this won't execute if the checks fail :) Could you add TODO? I hope we will solve this in the future

Reply:
It does since SuperConnector is `AutoClosable`

Reply:
How? `AutoClosable` is an interface to work with Java's `try-with-resources` syntax

Reply:
idk its a magic, but works
Try it out!

Reply:
```scala
class Resource extends AutoCloseable {
  override def close(): Unit = println("resource closed")
}

println("Start")
for {
  _ <- Either.cond(test = true, (), "1")
  r <- Right(new Resource)
  _ <- Either.cond(test = false, (), "2")
  _ <- Right(r.close())
} yield ()
println("End")
```

prints in Ammonite:
```
defined class Resource
Start
res1: scala.util.Either[String,Unit] = Left(2)
End
```

Reply:
gRPC client closed anyway, we can inspect this issue in another task